tests/components/modbus/test_light.py

```diff
@@ -307,21 +347,336 @@ async def test_light_service_turn(
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                    }
+                ]
+            },
+            155,
+            3000,
+            None,
+            None,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_COLOR_TEMP_REGISTER: 2,
+                    }
+                ]
+            },
+            128,
+            3000,
+            None,
+            20,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_BRIGHTNESS_REGISTER: 1,
+                        CONF_COLOR_TEMP_REGISTER: 2,
+                    }
+                ]
+            },
+            128,
+            2000,
+            50,
+            0,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_BRIGHTNESS_REGISTER: 1,
+                    }
+                ]
+            },
+            128,
+            None,
+            50,
+            None,
+        ),
+    ],
+)
+async def test_color_temp_brightness_light(
+    hass,
+    mock_modbus_ha,
+    brightness_input,
+    color_temp_input,
+    expected_brightness,
+    expected_color_temp,
+):
+    """Test Modbus Light color temperature and brightness."""
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+
+    service_data = {ATTR_ENTITY_ID: ENTITY_ID}
+
+    if brightness_input is not None:
+        service_data[ATTR_BRIGHTNESS] = brightness_input
+
+    if color_temp_input is not None:
+        service_data[ATTR_COLOR_TEMP_KELVIN] = color_temp_input
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        service_data=service_data,
+        blocking=True,
+    )
+    assert hass.states.get(ENTITY_ID).state == STATE_ON
+    calls = mock_modbus_ha.write_register.call_args_list
+    if expected_brightness is not None:
+        assert any(
+            call.args[0] == 1 and call.kwargs["value"] == expected_brightness
+            for call in calls
+        )
+    if expected_color_temp is not None:
+        assert any(
+            call.args[0] == 2 and call.kwargs["value"] == expected_color_temp
+            for call in calls
+        )
```

Comment:
could we also move these into the test params to remove the conditions? even if we need to remove the one without brightness or color temp